clients/v1_dbrps/v1_dbrps.go

```diff
@@ -0,0 +1,246 @@
+	clients.OrgParams
+	BucketID string
+	Default_ bool
+	DB       string
+	RP       string
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	if !params.OrgID.Valid() && params.OrgName == "" && c.ActiveConfig.Org == "" {
+		return clients.ErrMustSpecifyOrg
+	}
+
+	reqBody := api.DBRPCreate{
+		BucketID:        params.BucketID,
+		Database:        params.DB,
+		RetentionPolicy: params.RP,
+	}
+
+	// For compatibility with the cloud API for creating a DBRP, an org ID must be
+	// provided. The ID will be obtained based on the org name if an
+	// org name is provided but no ID is.
+	if params.OrgID.Valid() {
+		reqBody.OrgID = api.PtrString(params.OrgID.String())
+	} else {
+		orgName := params.OrgName
+		if orgName == "" {
+			orgName = c.ActiveConfig.Org
+		}
+		res, err := c.GetOrgs(ctx).Org(orgName).Execute()
+		if err != nil {
+			return fmt.Errorf("failed to look up ID for org %q: %w", orgName, err)
+		}
+		if len(res.GetOrgs()) == 0 {
+			return fmt.Errorf("no org found with name %q", orgName)
+		}
+		reqBody.OrgID = api.PtrString(res.GetOrgs()[0].GetId())
+	}
+
+	dbrp, err := c.PostDBRP(ctx).DBRPCreate(reqBody).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to create dbrp for bucket %q: %w", params.BucketID, err)
+	}
+
+	c.printDBRPs(dbrpPrintOpts{dbrp: &dbrp})
+
+	return nil
+}
+
+type UpdateParams struct {
+	clients.OrgParams
+	ID       string
+	Default_ bool
+	RP       string
+}
+
+func (c Client) Update(ctx context.Context, params *UpdateParams) error {
+	if !params.OrgID.Valid() && params.OrgName == "" && c.ActiveConfig.Org == "" {
+		return clients.ErrMustSpecifyOrg
+	}
+
+	reqBody := api.DBRPUpdate{}
+	if params.RP != "" {
+		reqBody.RetentionPolicy = &params.RP
+	}
+	if params.Default_ {
+		reqBody.Default = &params.Default_
+	}
+
+	req := c.PatchDBRPID(ctx, params.ID)
+	if params.OrgID.Valid() {
+		req = req.OrgID(params.OrgID.String())
+	}
+	if params.OrgName != "" {
+		req = req.Org(params.OrgName)
+	}
+	if !params.OrgID.Valid() && params.OrgName == "" {
+		req = req.Org(c.ActiveConfig.Org)
+	}
+
+	dbrp, err := req.DBRPUpdate(reqBody).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to update DBRP mapping %q: %w", params.ID, err)
+	}
+
+	dbrpContent := dbrp.GetContent()
+	c.printDBRPs(dbrpPrintOpts{dbrp: &dbrpContent})
+
+	return nil
+}
+
+type DeleteParams struct {
+	clients.OrgParams
+	ID string
+}
+
+func (c Client) Delete(ctx context.Context, params *DeleteParams) error {
+	if !params.OrgID.Valid() && params.OrgName == "" && c.ActiveConfig.Org == "" {
+		return clients.ErrMustSpecifyOrg
+	}
+
+	// Get the DBRP to verify that it exists, and to be able to print the results
+	// of the delete command, which output the details of the deleted DBRP.
+	getReq := c.GetDBRPsID(ctx, params.ID)
+	deleteReq := c.DeleteDBRPID(ctx, params.ID)
+	if params.OrgID.Valid() {
+		getReq = getReq.OrgID(params.OrgID.String())
+		deleteReq = deleteReq.OrgID(params.OrgID.String())
```

Comment:
I'm surprised either of these requests need org information to work. Can we double-check that the `org` and `orgId` params are actually used on the server-side in OSS and Cloud?

A common bug I've seen in the old CLI is that somebody copy-pasted the org/org-id flags into their command but never actually used the values. If that's the case here, we can update our CLI-parsing layer to keep the flags but mark them as hidden.

Reply:
You do need the `org` or `orgID` parameter for OSS `GET/{id}` and `DELETE/{id}` requests. I believe it's because of how the authorization is set up. I suspect it wouldn't _need_ to be set up like that - I'd think the authz could work based on the org information of the requested DBRP, since that object will have an organization ID and then it should be able to auth the request against that. Might require a bit more investigation as to why it works like it does...I opened an issue in`influxdb` regarding this: https://github.com/influxdata/influxdb/issues/21731

Interestingly, requests to cloud don't require the org info for `GET/{id}` or `DELETE/{id}` I don't know if this is actually intentional since the code looks to be the same. I think it's probably how the org information gets set on the request context in IDPE for every request even if it's not provided as a query parameter.